pkg/lib/cockpit-components-dropdown.tsx

```diff
@@ -35,49 +34,110 @@ import { EllipsisVIcon } from '@patternfly/react-icons';
 
+    const [drilledInMenus, setDrilledInMenus] = React.useState<string[]>([]);
+    const [drilldownItemPath, setDrilldownItemPath] = React.useState<string[]>([]);
+    const [menuHeights, setMenuHeights] = React.useState<{[id: string]: number}>({});
+    const [activeMenu, setActiveMenu] = React.useState<string>(id || "");
+
+    function onDrillIn(_event: React.KeyboardEvent | React.MouseEvent, fromId: string, toId: string, pathId: string) {
+        setDrilledInMenus([...drilledInMenus, fromId]);
+        setDrilldownItemPath([...drilldownItemPath, pathId]);
+        setActiveMenu(toId);
+    }
+
+    function onDrillOut(_event: React.KeyboardEvent | React.MouseEvent, toId: string) {
+        const drilledInMenusSansLast = drilledInMenus.slice(0, drilledInMenus.length - 1);
+        const pathSansLast = drilldownItemPath.slice(0, drilldownItemPath.length - 1);
+        setDrilledInMenus(drilledInMenusSansLast);
+        setDrilldownItemPath(pathSansLast);
+        setActiveMenu(toId);
+    }
+
+    function resetDrilldown() {
+        setDrilledInMenus([]);
+        setDrilldownItemPath([]);
+        setActiveMenu(id || "");
+    }
+
+    function onGetMenuHeight(menuId: string, height: number) {
+        if (menuHeights[menuId] === undefined || (menuId !== id && menuHeights[menuId] !== height)) {
+            setMenuHeights({ ...menuHeights, [menuId]: height });
+        }
+    }
+
+    function onSelect(_event: React.MouseEvent<Element, MouseEvent> | undefined, itemId: string | number | undefined) {
+        if (!itemId || typeof itemId == "number" || !itemId.startsWith("drilldown:")) {
+            setKebabOpen(false);
+            resetDrilldown();
+        }
+    }
+
     return (
         <Dropdown
+            {... id ? { id } : { } }
             onOpenChange={isOpen => setKebabOpen(isOpen)}
-            onSelect={() => setKebabOpen(false)}
+            onSelect={onSelect}
             toggle={(toggleRef) => (
                 <MenuToggle
                     id={toggleButtonId}
                     isDisabled={isDisabled}
                     ref={toggleRef}
                     variant="plain"
-                    onClick={() => setKebabOpen(!isKebabOpen)}
+                    onClick={() => {
+                        setKebabOpen(!isKebabOpen);
+                        resetDrilldown();
+                    }}
                     isExpanded={isKebabOpen}
                 >
                     <EllipsisVIcon />
                 </MenuToggle>
             )}
             isOpen={isKebabOpen}
-            popperProps={{ position }}
+            popperProps={{ position, ...popperProps }}
+            {... (containsDrilldown
+                ? {
+                    containsDrilldown: true,
+                    drilldownItemPath,
+                    drilledInMenus,
+                    onDrillIn,
+                    onDrillOut,
+                    activeMenu,
+                    onGetMenuHeight,
```

Comment:
I don't see this property documented here https://v5-archive.patternfly.org/components/menus/dropdown#dropdown ?

Reply:
It's there in PF6.. so I guess they just forgot.

Reply:
It's a property of [`Menu`](https://v5-archive.patternfly.org/components/menus/menu#menu) which Dropdown inherits from.